articles/making-api-requests-using-retrofit-android/index.md

```diff
@@ -0,0 +1,204 @@
+
+Add internet permission to you application.
+
+```Xml
+<uses-permission android:name="android.permission.INTERNET" />
+```
+
+#### Step 3 – Designing the UI for our application.
+In this step, we will design the layout for our application. Since this is a simple application, we will only use a `ListView` to display the API's information.
+Add the following lines of code to your layout resource file:
+
+```Xml
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MainActivity"
+    android:background="@color/white">
+
+    <ListView
+        android:id="@+id/superListView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content" />
+
+</LinearLayout>
+```
+
+### Step 4 - Create a model class.
+Next, we will create a model class that will contain the objects from the JSON. For our instance, we only want to get the names of the fictional characters.
+In the java directory, right-click and select new-→ java class-→ app/src/main/java--> class. We will name our model class **Results**. Add the following lines of code in `Results.java`:
+
+```Java
+public class Results {
+
+    @SerializedName("name")
+    private String superName;
+
+
+    public Results(String name) {
+        this.superName = name;
+    }
+
+    public String getName() {
+        return superName;
+    }
+}
+```
+**Note: The `SerializedName` annotation should always display the exact name of an object in the JSON file.**
+
+### Step 5 - Create a Retrofit instance.
+This java class is used to send requests to an API.  We specify the URL which contains the data required and use the `Retrofit Builder` class.
+In the java directory, right-click and select new-→ java class-→ app/src/main/java-→class. We shall name our class, **RetrofitClient**. Add the following lines of code to the `RetrofitClient.java`:
+
+```Java
+public class RetrofitClient {
+
+    private static RetrofitClient instance = null;
+    private Api myApi;
+
+    private RetrofitClient() {
+        Retrofit retrofit = new Retrofit.Builder().baseUrl(Api.BASE_URL)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build();
+        myApi = retrofit.create(Api.class);
+    }
+
+    public static synchronized RetrofitClient getInstance() {
+        if (instance == null) {
+            instance = new RetrofitClient();
+        }
+        return instance;
+    }
+
+    public Api getMyApi() {
+        return myApi;
+    }
+}
+```
+### Step 6 – Define the Endpoints.
+Endpoints usually are defined inside an **Interface** with special annotations either for the request method or the parameters. In our case, we will be using the `@GET` annotation since we are making a Get request. Our return value will be a `call<results>` object.
```

Comment:
Please rephrase this sentence. It seems hard to read.

Reply:
Done check it out.